### ISSUE TITLE
Using Expo.AuthSession works inconsistently and crashes the app

### ISSUE THREAD

**Issue by kuzzmi** [expo/expo#4201]
### Environment

`expo diagnostics` returns an error:

```
Cannot read property '1' of null
```

App's targets are iOS and Android standalone applications. 

### Steps to Reproduce

We're using Expo.AuthSession for external provider authentication. And there are two issues I'm experiencing with Expo.AuthSession:

(1) After a cold start of the app on Android, when in-app browser opens the page for auth.expo.io, page loading hangs on ~20%. This gets resolved only if you open the page, that Expo is trying to open, in the external browser on a smartphone and then repeat authentication flow.

(2) After getting back from in-app browser to the app again with tokens (after signing in through Expo.AuthSession) the app crashes. 

These functions are used to authenticate a user:

```
   this.login = provider => {
            let redirectUrl = AuthSession.getRedirectUrl();
            AuthSession.startAsync({
                authUrl: `${REACT_NATIVE_API_BASE}/api/auth/with_${provider}/?redirect_uri=${redirectUrl}`,
            }).then(({ params }) => {
                if (!params) {
                    return;
                }

                const { access_token } = params;
                this.props.saveTokens({
                    access_token,
                    refresh_token: '',
                });

                this.setState(state => ({
                    ...state,
                    redirected: true,
                }), () => {
                    setTimeout(() => {
                        this.props.history.push('/');
                    }, 500);
                });
            }).catch(e => {
                console.error(e);
            });
        };

        this.loginGoogle = () => {
            this.login('google');
        };

        this.loginFacebook = () => {
            this.login('facebook');
        };
```

### Expected Behavior

When the `Expo.AuthSession.startAsync` is used, the in-app browser doesn't get stuck and the app doesn't crash after retreiving tokens.

### Reproducible Demo

TBD

**Comment by DianaKoenraadt** [comment]
Similar issue here: I have a working app in development, but when I build a standalone app with `expo build:android`, the app crashes at `AuthSession.startAsync`.

**Comment by Jevon-Rhino** [comment]
Having the same issue, Anyone figured out a fix yet?

**Comment by DianaKoenraadt** [comment]
AuthSession is built on top of WebBrowser. So I ditched it and used WebBrowser directly.

**Comment by dbclkclk** [comment]
same issue here!!!

**Comment by sjchmiela** [comment]
We have published `expo-web-browser` to `5.0.3` with fixes. Please use `expo install expo-web-browser` to add the package to your dependencies and then ensure that the version is at least `5.0.3` (so replace `~5.0.0` with `~5.0.3` in your `package.json`).
